packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -558,32 +581,52 @@ class PubspecYaml {
     // Only if there were no dependencies: or dev_dependencies: sections could
     // we get here with this still null, and we should not have any such files
     // in our repo.
+    assert(endOfDirectDependencies != null);
     assert(lastPossiblePlace != null);
 
     // Now include all the transitive dependencies.
     final List<String> transitiveDependencyOutput = <String>[]; // the block of text to insert
+    final List<String> transitiveDevDependencyOutput = <String>[];
     final Set<String> transitiveDependencies = new Set<String>(); // which dependencies we need to handle
+    final Set<String> transitiveDevDependencies = new Set<String>();
+    
     // Merge the list of dependencies we've seen in this file and the dependencies we know this
     // file mentions that are already pinned (and which didn't get special processing above).
```

Comment:
all the comments here (this one, the next one, the next one after that, etc) all need updating to make sense now that there's two lists